framework/src/meshgenerators/SideSetExtruderGenerator.C

```diff
@@ -0,0 +1,100 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "SideSetExtruderGenerator.h"
+#include "CastUniquePointer.h"
+
+registerMooseObject("MooseApp", SideSetExtruderGenerator);
+
+InputParameters
+SideSetExtruderGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addClassDescription("Takes a 1D or 2D mesh and extrudes the entire structure along the "
+                             "specified axis increasing the dimensionality of the mesh.");
+
+  // list params
+  params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
+  params.addRequiredParam<RealVectorValue>("extrusion_vector",
+                                           "The direction and length of the extrusion");
+  params.addParam<unsigned int>("num_layers", 1, "The number of layers in the extruded mesh");
+  params.addRequiredParam<BoundaryName>("sideset",
+                                        "The side set (boundary) that will be extruded from");
+
+  return params;
+}
+
+SideSetExtruderGenerator::SideSetExtruderGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _original_input(getParam<MeshGeneratorName>("input")),
+    _extrusion_vector(getParam<RealVectorValue>("extrusion_vector")),
+    _num_layers(getParam<unsigned int>("num_layers")),
+    _sideset_name(getParam<BoundaryName>("sideset"))
+{
+  // constants needed only temporarily
+  const BoundaryName _EXTRUDED_BLOCK_NAME = "extruded_block";
+  const BoundaryName _SIDESET_TO_BE_STITCHED = "to_be_stitched";
```

Comment:
use regular moose naming convention:
extruded_block_name 
No capitals, no leading underscore since not a class member